src/main/java/core/basesyntax/service/impl/ShopDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ShopDao;
+import java.util.Map;
+import java.util.Set;
+
+public class ShopDaoImpl implements ShopDao {
+    private final Storage storage;
+
+    public ShopDaoImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public Map.Entry<String, Integer> get(String fruit) {
+        Set<Map.Entry<String, Integer>> fruits = storage.getShopStorage().entrySet();
+        for (Map.Entry<String, Integer> entry : fruits) {
+            if (entry.getKey().equals(fruit)) {
+                return entry;
+            }
+        }
+        throw new RuntimeException("There is no such fruit in the Storage");
+    }
```

Comment:
No logic in Dao , only basic operation CRUD 
create, update, update, delete.
And why you don't use map.get(fruit)?